diagnostics-app/src/app/core/services/routine-v2.service.ts

```diff
@@ -76,11 +76,12 @@ export class RoutineV2Service {
           (response: Response) => {
             if (!response.success) {
               return reject(response.error);
-            } else if (!response.routineV2) {
-              throw 'Invalid response';
-            } else {
+            }
+            // Some APIs have response, stored in the routineV2 field.
+            if (response.routineV2) {
               return resolve(response.routineV2);
             }
+            return resolve(null);
```

Comment:
`return resolve(response!.routineV2);`

Reply:
Done. 

Since `response?.routineV2` returns undefined when the field does not exist, but the codebase uses null to pass around intentionally empty objects, I've typecasted it into null via the Nullish coalescing operator (??).